### ISSUE TITLE
Bug: `generateMarkdown`'s `alias` parameter has no effect for non-markdown files

### ISSUE THREAD

**Issue by RyotaUshio** [obsidianmd/obsidian-api#154]
**Steps to reproduce:**
- In the sandbox vault, **turn off "use [[wikilinks]]"**.
- Add an attachment file such as PNG or PDF, say `file.png`.
- Open the dev console and run the following:
  ```js
  const file = app.vault.getAbstractFileByPath('file.png');
  console.log(app.fileManager.generateMarkdownLink(file, '', undefined, 'Display text'));
  ```

**Expected result:**

`![Display text](file.png)`

**Actual result:**

`![](file.png)`

The `alias` parameter has no effect.

**Comment by RyotaUshio** [comment]
Now I read the source code from the dev tool, and it seems like it's intended.

I guess this design is because `generateMarkdownLink` generates an embed link for attachment files and display text is not important for embeds.

However, `generateMarkdownLink` is also used to generate a non-embed link like so:

```ts
app.fileManager.generateMarkdownLink(...).slice(1)
```

I personally do this for generating links for PDF files, and even Obsidian itself does the same thing in many places.

So I believe it will be beneficial to let `alias` work for attachments.

**Comment by RyotaUshio** [comment]
For example, if I turn on "Use [[Wikilinks]]" and copy a link to a PDF annotation, this becomes problematic:

https://github.com/obsidianmd/obsidian-api/assets/72342591/736b51c0-8919-43f0-a17d-59d2a28f54dd

**Comment by liamcain** [comment]
Thanks, this will be fixed in v1.7
